[pt-br] Add tasks/debug/debug-application/debug-pods (#49678)
kubernetes/website: content/pt-br/docs/tasks/debug/debug-application/debug-pods.md

  Consulte o documento [Recursos de Computação](/docs/concepts/configuration/manage-resources-containers/) para mais informações.

* **Você está usando `hostPort`**: Quando você vincula um Pod a um `hostPort`, há um número limitado de locais onde esse pod pode ser agendado.
  Na maioria dos casos, `hostPort` é desnecessário, tente usar um objeto Service para expor seu Pod.
  Se você realmente precisar de `hostPort`, então só poderá agendar tantos Pods quanto o número de nós no seu cluster Kubernetes.

#### Meu pod fica em estado waiting

Se um Pod estiver preso no estado `Waiting`, significa que ele foi agendado para um nó de trabalho,
mas não pode ser executado nessa máquina. Novamente, as informações do comando `kubectl describe ...`
devem fornecer detalhes úteis.

A causa mais comum para Pods em estado `Waiting` é a falha ao puxar a imagem.
Há três coisas que você deve verificar:

* Certifique-se de que o nome da imagem está correto.
* Você enviou a imagem para o registro?
* Tente puxar a imagem manualmente para verificar se ela pode ser baixada. Por exemplo,
  se você usa Docker no seu PC, execute `docker pull <image>`.

#### Meu pod fica em estado terminating

> Se um Pod estiver preso no estado `Terminating`, significa que uma solicitação de exclusão foi emitida,
> mas o control plane não conseguiu remover o objeto do Pod.
> 
> Isso geralmente ocorre se o Pod possui um [finalizer](/docs/concepts/overview/working-with-objects/finalizers/)
> e há um [admission webhook](/docs/reference/access-authn-authz/extensible-admission-controllers/)
> instalado no cluster que impede o control plane de remover o finalizer.
> 
> Para identificar esse cenário, verifique se seu cluster possui algum ValidatingWebhookConfiguration ou MutatingWebhookConfiguration que tenha como alvo
> operações `UPDATE` para recursos `pods`.
> 
> Se o webhook for fornecido por um terceiro:
> - Certifique-se de estar usando a versão mais recente.
> - Desative o webhook para operações `UPDATE`.
> - Relate um problema ao provedor correspondente.
> 
> Se você for o autor do webhook:
> - Para um webhook mutante, certifique-se de que ele nunca altere campos imutáveis
>   em operações `UPDATE`. Por exemplo, mudanças em containers geralmente não são permitidas.
> - Para um webhook validador, garanta que suas regras de validação se apliquem apenas a novas alterações.
>   Em outras palavras, você deve permitir que Pods com violações existentes passem pela validação.
>   Isso permite que Pods criados antes da instalação do webhook continuem em execução.

Reviewer: * Camada de gerenciamento é uma tradução que já temos padronizada para o termo control plane: https://kubernetes.io/pt-br/docs/contribute/localization_pt-br/#dicion%C3%A1rio-de-termos-com-tradu%C3%A7%C3%A3o
* Embora a tradução literal de mutating webhook seja webhook mutante e de validating webhook seja webhook validador, gostaria de sugerir o uso de webhook de mutação e webhook de validação, já que "mutating"/"validating" aqui se referem ao tipo de operação que o webhook vai realizar nos objetos.
* Na localização em português, padronizamos a aportuguesação do termo "container" em "contêiner".

```suggestion
Se um Pod estiver preso no estado `Terminating`, significa que uma solicitação de exclusão foi emitida,
mas a camada de gerenciamento não conseguiu remover o objeto do Pod.

Isso geralmente ocorre se o Pod possui um [finalizer](/docs/concepts/overview/working-with-objects/finalizers/)
e há um [admission webhook](/docs/reference/access-authn-authz/extensible-admission-controllers/)
instalado no cluster que impede a camada de gerenciamento de remover o finalizer.

Para identificar esse cenário, verifique se seu cluster possui algum ValidatingWebhookConfiguration ou MutatingWebhookConfiguration que tenha como alvo
operações `UPDATE` para recursos `pods`.

Se o webhook for fornecido por um terceiro:
- Certifique-se de estar usando a versão mais recente.
- Desative o webhook para operações `UPDATE`.
- Relate um problema ao provedor correspondente.

Se você for o autor do webhook:
- Para um webhook de mutação, certifique-se de que ele nunca altere campos imutáveis
  em operações `UPDATE`. Por exemplo, mudanças em contêineres geralmente não são permitidas.
- Para um webhook de validação, garanta que suas regras de validação se apliquem apenas a novas alterações.
  Em outras palavras, você deve permitir que Pods com violações existentes passem pela validação.
  Isso permite que Pods criados antes da instalação do webhook continuem em execução.
```

Reviewer: Oi @paulofponciano! Parece que este item ainda ficou inalterado.

> #### Meu pod está falhando ou não está saudável
> 
> Depois que seu pod for agendado, você pode usar os métodos descritos em
> [Depurando Pods em Execução](/docs/tasks/debug/debug-application/debug-running-pod/) para depuração.

Reviewer: Sugestão: em alguns lugares na documentação usamos "íntegro" ao invés de "saudável" (já que se trata de algo inanimado).

```suggestion
#### Meu pod está falhando ou não está íntegro

Depois que seu Pod for alocado, você pode usar os métodos descritos em
[Depurando Pods em Execução](/docs/tasks/debug/debug-application/debug-running-pod/) para depuração.
```

> #### Meu pod está em execução, mas não faz o que eu defini

Reviewer: Sugestão para esta seção: usar "servidor da API" no lugar de apiserver.

A primeira coisa a fazer é excluir seu pod e tentar criá-lo novamente usando a opção `--validate`.
Por exemplo, execute `kubectl apply --validate -f mypod.yaml`.
Se você digitou `command` incorretamente como `commnd`, verá um erro como este:

```shell
I0805 10:43:25.129850   46757 schema.go:126] unknown field: commnd
I0805 10:43:25.129973   46757 schema.go:129] this may be a false alarm, see https://github.com/kubernetes/kubernetes/issues/6842
pods/mypod
```

<!-- TODO: Now that #11914 is merged, this advice may need to be updated -->

A próxima coisa a verificar é se o pod no apiserver corresponde ao pod que você pretendia criar
(por exemplo, no arquivo yaml em sua máquina local).
Por exemplo, execute `kubectl get pods/mypod -o yaml > mypod-on-apiserver.yaml` em seguida,
compare manualmente a descrição original do pod, `mypod.yaml` com a versão obtida do apiserver, `mypod-on-apiserver.yaml`.  
Normalmente, a versão do "apiserver" terá algumas linhas extras que não estão na versão original,
o que é esperado. No entanto, se houver linhas na versão original que não aparecem na versão do apiserver, 
isso pode indicar um problema na especificação do seu pod.

> ### Depurando Replication Controllers

Reviewer: Para que o link da seção "Diagnosticando o problema" funcione corretamente:

```suggestion
### Depurando Replication Controllers {#debugging-replication-controllers}
```

Replication Controllers são bastante diretos. Eles podem criar pods ou não.
Se não conseguirem criar pods, consulte as
[instruções acima](#debugging-pods) para depurar seus pods.

> Você também pode usar `kubectl describe rc ${CONTROLLER_NAME}` para introspectar eventos

Reviewer: Não consegui encontrar a palavra "introspectar" no dicionário de português brasileiro. Talvez "examinar" seja uma boa tradução?

```suggestion
Você também pode usar `kubectl describe rc ${CONTROLLER_NAME}` para examinar eventos
```

relacionados ao replication controller.

> ### Depurando Services

Reviewer: Para que o link da seção "Diagnosticando o problema" funcione corretamente:

```suggestion
### Depurando Services {#debugging-services}
```
